src/k2/module/k23si/Module.cpp

```diff
@@ -272,6 +277,285 @@ bool K23SIPartitionModule::_validateStaleWrite(dto::K23SIWriteRequest& request,
+                            fieldsOffset.push_back(tmpOffset);
+                            break;
+                        }
+                        case dto::FieldType::UINT64T: {
+                            uint32_t tmpOffset = fieldsOffset[baseCursor] + sizeof(uint64_t);
+                            fieldsOffset.push_back(tmpOffset);
+                            break;
+                        }
+                        default :
+                            return false;
+                        } // end switch
+                    }else {
+                        fieldsOffset.push_back(fieldsOffset[baseCursor]);
+                    }
+                }
+                
+                if (request.value.excludedFields[i] == false) _advancePayloadPosition(request.value.fieldData, schema.fields[i].type); // have to do first
+                if (versions[0].value.excludedFields.empty() || versions[0].value.excludedFields[findField] == false) {
+                    // copy value from base
+                    basePayload.seek(fieldsOffset[findField]);
+                    if (!_copyPayloadBaseToUpdate(basePayload, payload, baseSchema.fields[findField].type)) return false;
+                    request.value.excludedFields[i] = false;
+                } else {
+                    // set excludedFields[i]=true
+                    request.value.excludedFields[i] = true;
+                }
+                
+                baseCursor = findField + 1;
+            }
+        } else {
+            // this field is to be updated.
+            if (request.value.excludedFields[i] == false) {
+                // request's payload has a value.
+                // 1. write() value from req's SKVRecord to payload
+                if (!_copyPayloadBaseToUpdate(request.value.fieldData, payload, schema.fields[i].type)) return false;
+            } else {
+                // request's payload skips this field
+                // 1. set excludedField
+                request.value.excludedFields[i] = true;
+            }
+        } 
+    }
+    
+    request.value.fieldData = std::move(payload);
+    request.value.fieldData.truncateToCurrent();
+    return true;
+}
+
+bool K23SIPartitionModule::_parsePartialRecord(dto::K23SIPartialUpdateRequest& request, std::deque<dto::DataRecord>& versions) {
+    auto schemaIt = _schemas.find(request.key.schemaName);
+    if (schemaIt == _schemas.end()) return false;
+    auto schemaVer = schemaIt->second.find(request.value.schemaVersion);
+    if (schemaVer == schemaIt->second.end()) return false;
+    dto::Schema schema = schemaVer->second;
+    
+    if (!request.value.excludedFields.size()) {
+        request.value.excludedFields = std::vector<bool>(schema.fields.size(), false);
+    }
+    
+    Payload basePayload = versions[0].value.fieldData.shareAll();   // base payload
+    Payload payload(Payload::DefaultAllocator);                     // payload for new record
```

Comment:
I think payload and basePayload are unused here and can be removed?

Reply:
yes, they should be removed.